api/internal/localizer/localizer_test.go

```diff
@@ -454,29 +502,15 @@ kind: Kustomization
 replacements:
 - path: replacement.yaml
 - source:
-    fieldPath: path
+    fieldPath: path.0
```

Comment:
why does this change?

Reply:
Not for any important reason. Down to revert it if the change is distracting.

I checked the `ReplacementTransformer` code and tried to cover the different formats that it accepts. I added this because I standardized `replacements` and `ReplacementTransformer` tests with [`replacementTransformerWithPath`](https://github.com/kubernetes-sigs/kustomize/pull/4943/files#diff-c21f7b3d0afda4a64163411e90e72d2fed6c3df14c7b6f8ed07237bb5e1fc91cR30) and [`replacements`](https://github.com/kubernetes-sigs/kustomize/pull/4943/files#diff-c21f7b3d0afda4a64163411e90e72d2fed6c3df14c7b6f8ed07237bb5e1fc91cR48), but neither of these use a non-negative index in `source`.